streams/src/test/java/org/apache/kafka/test/MockStreamsMetrics.java

```diff
@@ -14,15 +14,17 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.kafka.streams.processor.internals;
+package org.apache.kafka.test;
 
 import java.util.Collections;
+
 import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.streams.processor.internals.StreamsMetricsImpl;
 
 public class MockStreamsMetrics extends StreamsMetricsImpl {
 
-    public MockStreamsMetrics(Metrics metrics) {
-        super(metrics, "mock-stream-metrics",
+    public MockStreamsMetrics() {
+        super(new Metrics(), "mock-stream-metrics",
```

Comment:
So far this mock object is still actually implementing the full functionalities of a `StreamsMetricsImpl` since the mock actually extends it; the "mocking" is that it is passing in a default group name and an empty tag map.

A real "mock" class, would be not implementing any real functionalities (so not needing an underlying `Metrics` object) but just implementing the `StreamsMetrics` interface as no-ops, or at best recording the times that some APIs has been called etc. For cases that really needs a functional `StreamsMetricsImpl` object, we should just pass it instead of the mocking object. Please see my comments in the related class.

Reply:
Ok let me try other way . I have to change lot of test cases . Thanks for the information :-)